smarts/core/agent_manager.py

```diff
@@ -127,6 +127,12 @@ def observe_from(self, sim, vehicle_ids):
             rewards[agent_id] = vehicle.trip_meter_sensor(increment=True)
             scores[agent_id] = vehicle.trip_meter_sensor()
 
+        # also add agents that were done in virtue of just dropping out
+        for dvid in done_this_step:
+            aid = self._vehicle_with_sensors.get(dvid, None)
+            if aid:
+                dones[aid] = True
```

Comment:
For `dvid` and `aid`, consider something like `done_v_id`, `a_id`, or `agent_id`, in order to be similar to those used a couple of lines above.